pkg/app/piped/cloudprovider/kubernetes/diff.go

```diff
@@ -57,6 +61,10 @@ func Diff(old, new Manifest, opts ...diff.Option) (*diff.Result, error) {
 			return nil, err
 		}
 	}
+
+	old.u = remarshal(old.u)
+	new.u = remarshal(new.u)
```

Comment:
Why do we need to remarshal this one?

Reply:
I think this is not needed(if I delete `new.u ...`, it works fine) but argoproj does like this. So I wrote like this.

Reply:
Got it. I guess re-marshal the new one to ensure that both of them are in the same struct model to be more comparable.
But let's go without that to see what happens. 😄

Reply:
Got it. I guess re-marshal the new one to ensure that both of them are in the same struct model to be more comparable.
But let's go without that to see what happens. 😄